### PR TITLE
mkdir: fix some permission modes that differ from GNU mkdir

### LINKED ISSUE #3645: mkdir: permissions when using -m differ from GNU
When using `-m` option to `mkdir`, the permissions of the created directory differ from those of GNU mkdir. I think this may have something to do with the umask.

GNU mkdir:
```sh
$ mkdir -m + d
$ ls -ld d
drwxrwxr-x 2 jeffrey jeffrey 4096 Jun 16 22:12 d
```

uutils mkdir
```sh
$ mkdir -m + d
$ ls -ld d
drwxr-xr-x 2 jeffrey jeffrey 4096 Jun 16 22:12 d
```

umask on my machine:
```sh
$ umask
0002
```

### PR DESCRIPTION
**Pull request by wan-nyan-wan**
Fixes #3645

### CODE REVIEW

File: src/uu/mkdir/src/mkdir.rs [hunk 1]
```diff
@@ -20,8 +20,6 @@ use uucore::mode;
 use uucore::{display::Quotable, fs::dir_strip_dot_for_creation};
 use uucore::{format_usage, InvalidEncodingHandling};
 
-static DEFAULT_PERM: u32 = 0o755;
```
Comment (sylvestre) [inline src/uu/mkdir/src/mkdir.rs:23]: windows is still using this variable ;)

**Comment by tertsdiepraam** [comment]
What was the part that you were unsure about?

File: src/uu/mkdir/src/mkdir.rs (hunk 1 above)
Comment (wan-nyan-wan) [reply src/uu/mkdir/src/mkdir.rs:23]: I didn't consider supporting windows. Well, how do I test windows behavior?

Comment (sylvestre) [reply src/uu/mkdir/src/mkdir.rs:23]: Most of the time, fixing the error in the CI is good enough
If you need to do more complex testing (probably not the case here), you could install a Windows VM https://developer.microsoft.com/windows/downloads/virtual-machines/

Comment (wan-nyan-wan) [reply src/uu/mkdir/src/mkdir.rs]: That's really helpful! I will use it as a reference.

**Comment by wan-nyan-wan** [comment]
@tertsdiepraam I implemented this PR with GNU mkdir and chmod behavior. Almost case, I tried to match result of uutils mkdir to GNU chmod because of I think permissions behavior in GNU mkdir may be incorrect.

**Comment by wan-nyan-wan** [comment]
For example, the below mkdir case is strange, I think. The permission should be `drwxr-xr--`.
How do you think these?
```:bash
$ umask
027
$ mkdir -m a+r d
$ ls -ld d
drwxrwxrwx - user 29 Jul 15:56 d1
```

**Comment by tertsdiepraam** [comment]
Oh interesting, I think this warrants a bit more investigation before we can merge this. I think it's unlikely to be a bug, though the difference with chmod is weird. Have you figured out what the difference is?

**Comment by tertsdiepraam** [comment]
So essentially, it starts with 777 and then removes the umask if none of `augo` are given? I agree that it's confusing, but I think we should keep compatibility.

**Comment by tertsdiepraam** [comment]
Excellent! Thanks!

**Pull request closed without merging**